bijection-finagle-mysql/src/test/scala/com/twitter/bijection/finagle_mysql/MySqlConversionLaws.scala

```diff
@@ -39,18 +37,23 @@ class MySqlConversionLaws extends PropSpec with PropertyChecks with MustMatchers
   implicit val floatValueArb = Arbitrary(arbitrary[Float].map(FloatValue.apply))
   implicit val doubleValueArb = Arbitrary(arbitrary[Double].map(DoubleValue.apply))
   implicit val stringValueArb = Arbitrary(arbitrary[String].map(StringValue.apply))
-  implicit val timestampValueArb = Arbitrary(arbitrary[Date].map(d => new Timestamp(d.getTime)))
   implicit val nullValueArb = Arbitrary(Gen.const(NullValue))
   implicit val emptyValueArb = Arbitrary(Gen.const(EmptyValue))
-  implicit val valueArb: Arbitrary[Value] = Arbitrary(Gen.oneOf(arbitrary[ByteValue],
-    arbitrary[NullValue.type],
-    arbitrary[EmptyValue.type],
-    arbitrary[ShortValue],
-    arbitrary[IntValue],
-    arbitrary[LongValue],
-    arbitrary[FloatValue],
-    arbitrary[DoubleValue],
-    arbitrary[StringValue]))
+
+  val timeGenerator = Gen.choose(1L, 253375661380264L) // until year 9999
```

Comment:
There were some overflows happening involving timestamps millions of years in the future. I've constrained this to only generate timestamps until year 9999, that solved the problem.